### DISCUSSION TITLE
Ensure order of configurations

### DISCUSSION THREAD

**Discussion by m3tomlins** [Dynatrace/dynatrace-configuration-as-code#1223] [General]
I read the accepted technique for ordering in the config.yaml file: https://www.dynatrace.com/support/help/manage/configuration-as-code/monaco/guides/order-of-configurations

But, if I have 1500 conditional-naming-service rules in the configuration - and I insert a new naming rule in position #5...

...I have to update all of the reference dependencies for 1495 other entries in the config.yaml.

Any suggestions? Could I write a separate automated script to generate the config.yaml IN ORDER without all that editing?

Wow it would be nice to have the "monaco download" give the reference order from the source tenant. :)

**Comment by UnseenWizzard** [comment]
Hi @m3tomlins,
currently, managing this manually is the only option. 

We aim to support ordering natively in the future, at least for Settings 2.0 configurations.

**Reply by m3tomlins** [reply]
Great! For now I just glued together a solution using parameter.order reference configurations into a bash script with yq. In our deploy folder I just have another file with the ordered list of config ids. The script parses the original config.yaml, adds the reference id parameter and spits out the new ordered config.yaml. That makes it easy to "insert" a new config into the ordered list and then regenerate the config.yaml again.

For actual deployment I just generate a deletefile from the ordered config.yaml, delete those configs and re-deploy. That way any new configurations in the tenant that are NOT auto-deployed from the Monaco pipeline will be left undisturbed, but alas - unordered.

**Reply by UnseenWizzard** [reply]
Great to hear you have a workaround! And thanks for sharing how you've achieved it so others might do the same!
